### DISCUSSION TITLE
PHOTOPRISM_DISABLE_CLASSIFICATION is not working

### DISCUSSION THREAD

**Discussion by human705** [photoprism/photoprism#4432] [Q&A]
Hi all, I'm running photoprism on docker and I'd like to disable the feature that adds keywords to my picture. I'd like to use only categories and titles assigned to the photo using XNView. I thought I could accomplish it by setting the 
`PHOTOPRISM_DISABLE_CLASSIFICATION: "true" ` flag but it doesn't seem to do what I need. I also tried setting 
`PHOTOPRISM_DISABLE_TENSORFLOW: "true"` just to see if that stops it but no luck. I'm attaching trace logs here. You can see a line similar to the one below for all the images that were added.

level=trace msg="index: Processed/2022/2022-04-30/Baseball_1077.jpg has keywords baseball, basketball, brown, gibbs, nesconset, new-york, pond, processed, road, smithtown, sports, usa"

TIA
[trace.log](https://github.com/user-attachments/files/16442321/trace.log)

**Comment by lastzero** [comment]
Keywords are also extracted from file and folder names. No AI/ML is required for this, and any titles specified in the metadata of your pictures will take precedence over the titles automatically generated by PhotoPrism. Could this just be a search issue, i.e. you don't want to find pictures that have the word you are searching for in their filename or e.g. a matching color (which might also be added to the automatically generated keywords, but not to the title)?

**Reply by human705** [reply]
I thought of that but the name and folder have none of the keywords assigned. I'm attaching the exiftool generated yaml file and the yaml file created by photoprism for one of the pictures in question. I'm not sure where the keywords in the photoprism generated yamf file are coming from. 

[Baseball_1074yml.txt](https://github.com/user-attachments/files/16444603/Baseball_1074yml.txt)
[exiftoolyml.txt](https://github.com/user-attachments/files/16444612/exiftoolyml.txt)

**Reply by human705** [reply]
After further review it appears that the keywords are added by the reverse geo lookup. I added `PHOTOPRISM_DISABLE_PLACES: "true"` and that added only 3 keywords (baseball, blue, processed) for the Baseball_1074.jpg photo. 

I also found that keywords **cannot** be removed once in the database. If I remove the gps metadata from my picture and do a "Complete Rescan" all keywords remain in the database. 

This behavior remains the same if keywords are added/removed from the details GUI of a photo. Anything removed will come back after a rescan. Anything added, stays.

I would suggest a check is added to the rescan option to allow the user to pick if existing keyword list should be preserved.

**Reply by lastzero** [reply]
This is because we do not currently record the history of each keyword and how it was added. However, if you know a little SQL or pay for technical support, it is possible to delete all keywords from the database with a single SQL statement.

**Reply by human705** [reply]
Thank you! Is there a link to the SQL schema that you can share?

**Reply by lastzero** [reply]
https://docs.photoprism.app/developer-guide/database/

**Reply by human705** [reply]
Thank you!
